### PR TITLE
Feature itembuilder texture layers

### PR DESCRIPTION
**Pull request by LLytho**
Implements #341

- Adds color callbacks for dynamically tint the texture layers
- Passing `null` to `.group()` will hide the item from creative tabs
- `.texture(layerName, texture)` now possible to specify the layer
- modelJson will now be used

```js
onEvent("item.registry", (event) => {
	/**
	 * Old style with just setting color by index still works!
	 */
	event
		.create("old_color_by_index")
		.textureJson({
			layer0: "minecraft:item/paper",
			layer1: "minecraft:item/ghast_tear",
		})
		.color(0, "#70F00F")
		.color(1, "#00FFF0");

	event
		.create("cooler_sword", "sword")
		.displayName("Test Cooler Sword")
		.texture("minecraft:item/iron_sword")
		.color((itemstack) => {
			/**
			 * Example by storing the color in the nbt of the itemstack
			 * You have to return -1 to apply no tint.
			 *
			 * U can test this through: /give @p kubejs:cooler_sword{color:"#ff0000"}
			 */
			if (itemstack.nbt && itemstack.nbt.color) {
				return itemstack.nbt.color;
			}

			return -1;
		});

	event
		.create("test_item")
		.displayName("Test Item")
		.textureJson({
			layer0: "minecraft:item/beef",
			layer1: "minecraft:item/ghast_tear",
		})
		.color((itemstack, tintIndex) => {
			/**
			 * If you want to apply the color to a specific layer, you can use the tintIndex
			 * tintIndex is the texture layer index from the model: layer0 -> 0, layer1 -> 1, etc.
			 * U can use the `Color` wrapper for some default colors
			 *
			 * This example will apply the color to the ghast_tear texture.
			 */
			if (tintIndex == 1) {
				return Color.BLUE;
			}
			return -1;
		});

	/**
	 * Set a texture for a specific layer
	 */
	event.create("test_sword", "sword").displayName("Test Sword").texture("layer0", "minecraft:item/bell");

	/**
	 * Directly set your custom model json
	 */
	event.create("test_something").displayName("Test something").modelJson({
		parent: "minecraft:block/anvil",
	});

	/**
	 * .group(null) to hide the item from creative tabs
	 */
	event.create("test_no_group").displayName("Test No Group").group(null);
});

```

### CODE REVIEW

File: common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java [hunk 1]
```diff
@@ -131,6 +134,13 @@ public void clientRegistry(Minecraft minecraft) {
 		if (!color.isEmpty()) {
 			ColorHandlerRegistry.registerItemColors((stack, index) -> color.get(index), this);
 		}
+
+		if(colorCallback != null) {
+			ColorHandlerRegistry.registerItemColors((itemStack, tintIndex) -> {
+				Object color = colorCallback.getColor(new ItemStackJS(itemStack), tintIndex);
+				return ColorWrapper.of(color).getRgbKJS();
+			}, this);
+		}
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java:143]: Could we somehow merge these two into the same property? I don't really like the thought of maintaining two completely separate ways of assigning colours to an item

**Review by MaxNeedsSnacks — CHANGES_REQUESTED** [review]

File: common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java (hunk 1 above)
Comment (LLytho) [reply common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java]: Makes totally sense to merge them

File: common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java [hunk 2]
```diff
@@ -249,4 +279,39 @@ public ItemBuilder modifyAttribute(ResourceLocation attribute, String identifier
 		attributes.put(attribute, new AttributeModifier(new UUID(identifier.hashCode(), identifier.hashCode()), identifier, d, operation));
 		return this;
 	}
+
+	public interface ItemColorJS {
+		Object getColor(ItemStackJS itemStack, int tintIndex);
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/mods/kubejs/item/ItemBuilder.java:284]: Could this use the Rhino `Color` type as a result, or does Rhino not type wrap the *result* of a method automatically?

**Review by MaxNeedsSnacks — APPROVED** [review]

**Comment by MaxNeedsSnacks** [comment]
Screw it I'll merge these and the other PRs *now* :P

**Pull request merged**